docs: update notebook for JSON subfields support in to_pandas_batches()

shuoweil opened the pull request:
When displaying a DataFrame containing JSON columns (including nested JSON in lists or structs), the `anywidget` table would fail to render and fall back to the "Computation deferred" message. This was caused by a limitation in PyArrow (apache/arrow#45262).
 
This branch only updates the relevant notebook. The underlying code fix for handling JSON subfields in STRUCT and ARRAY columns was already merged via PR #2216 (fix: support results with STRUCT and ARRAY columns containing JSON subfields in to_pandas_batches()).

This change simply aligns the notebook with the updated code.

Fixes #<448126500 and 453561268> 🦕

shuoweil committed:
perf: Default to interactive display for SQL in anywidget mode

Previously, SQL queries in anywidget mode would fall back to deferred execution, showing a dry run instead of an interactive table.

This change modifies the display logic to directly use the anywidget interactive display for SQL queries, providing a more consistent and responsive user experience. A test case has been added to verify this behavior.

shuoweil committed:
fix: resolve double printing issue in anywidget mode

tswast reviewed (changes requested):
In b/448126500 I suggest investigating the errors that happen when visualizing STRUCT columns, but this PR doesn't test such cases.

shuoweil committed:
change to ai.generate

shuoweil committed:
feat: Add test case for STRUCT column in anywidget

Adds a test case to verify that a DataFrame with a STRUCT column is
correctly displayed in anywidget mode.

This test confirms that displaying a STRUCT column does not raise an
exception that would trigger the fallback to the deferred representation.
It mocks `IPython.display.display` to capture the `TableWidget` instance
and asserts that the rendered HTML contains the expected string
representation of the STRUCT data.

shuoweil committed:
fix presubmit

tswast commented on bigframes/dataframe.py:
```diff
@@ -781,11 +781,23 @@ def __repr__(self) -> str:
+
+                widget = display.TableWidget(self.copy())
+                return widget._repr_html_()  # Return widget's HTML representation
+            except (AttributeError, ValueError, ImportError):
```
If there are cases where we raise AttributeError or ValueError, I would like to know about it rather than hide bugs. Regarding the ImportError, perhaps we can be more explicit about when anywidget is not installed?

tswast commented on bigframes/dataframe.py:
```diff
@@ -783,11 +783,23 @@ def __repr__(self) -> str:
+                from bigframes import display
+
+                widget = display.TableWidget(self.copy())
+                return widget._repr_html_()  # Return widget's HTML representation
```
We shouldn't be returning HTML in `__repr__`.

shuoweil committed:
Revert accidental changes to test_function.py

shuoweil commented on bigframes/operations/blob.py:
```diff
@@ -804,35 +804,27 @@ def audio_transcribe(
-            model_name=model_name,
-            session=self._block.session,
-            connection_name=connection,
+        # Use bbq.ai.generate() to transcribe audio
```
It is true. I will revert the change.

shuoweil committed:
revert accidental change to blob.py

shuoweil committed:
change return type

tswast commented on bigframes/dataframe.py:
```diff
@@ -783,9 +782,9 @@ def __repr__(self) -> str:
         max_results = opts.max_rows
-        # anywdiget mode uses the same display logic as the "deferred" mode
-        # for faster execution
-        if opts.repr_mode in ("deferred", "anywidget"):
```
Let's revert this.

shuoweil committed:
add todo and revert change

shuoweil committed:
Revert "add todo and revert change"

This reverts commit 153e1d203c273d6755623b3db30bd2256a240cc1.

shuoweil committed:
Add todo

shuoweil committed:
Fix: Handle JSON dtype in anywidget display

This commit fixes an AttributeError that occurred when displaying a
DataFrame with a JSON column in anywidget mode. The dtype check
was incorrect and has been updated. Additionally, the SQL compilation
for casting JSON to string has been corrected to use TO_JSON_STRING.

shuoweil committed:
revert a change

shuoweil committed:
revert a change

shuoweil committed:
Revert: Restore bigframes/dataframe.py to state from 42da847

shuoweil committed:
remove anywidget from early return, allow execution proceeds to _repr_html_()

shuoweil committed:
remove unnecessary changes

shuoweil committed:
remove redundant code change

shuoweil committed:
code style change

shuoweil committed:
tescase update

shuoweil committed:
revert a change

shuoweil committed:
final touch of notebook

shuoweil committed:
fix presumbit error

shuoweil committed:
remove invlaid test with anywidget bug fix

shuoweil committed:
fix presubmit

shuoweil committed:
fix polar complier

shuoweil committed:
Revert an unnecessary change

shuoweil committed:
apply the workaround to i/O layer

shuoweil committed:
Revert scalar_op_registry.py chnage

shuoweil committed:
remove unnecessary import

shuoweil committed:
Remove duplicate conversation

shuoweil committed:
revert changes to test_dataframe.py

shuoweil committed:
notebook update

shuoweil committed:
Merge branch 'main' into shuowei-anywidget-col

shuoweil committed:
call API on local data for complier.py

shuoweil committed:
add more testcase

shuoweil committed:
modfiy polars import

shuoweil committed:
Merge branch 'main' into shuowei-anywidget-col

shuoweil committed:
fix failed tests

jialuoo committed:
chore: Migrate minimum_op operator to SQLGlot (#2205)

jialuoo committed:
chore: Migrate round_op operator to SQLGlot (#2204)

This commit migrates the `round_op` operator from the Ibis compiler to the SQLGlot compiler.

shuoweil committed:
fix: Improve error handling in blob operations (#2194)

* add error handling for audio_transcribe

* add error handling for pdf functions

* add eror handling for image functions

* final touch

* restore rename

* update notebook to better reflect our new code change

* return None on error with verbose=False for image functions

* define typing module in udf

* only use local variable

* Refactor code

tswast committed:
refactor: update geo "spec" and split geo ops in ibis compiler (#2208)

tswast committed:
feat: support INFORMATION_SCHEMA views in `read_gbq` (#1895)

* feat: support INFORMATION_SCHEMA tables in read_gbq

* avoid storage semi executor

* use faster tables for peek tests

* more tests

* fix mypy

* Update bigframes/session/_io/bigquery/read_gbq_table.py

* immediately query for information_schema tables

* Fix mypy errors and temporarily update python version

* snapshot

* snapshot again

shuoweil committed:
Merge branch 'main' into shuowei-anywidget-col

shuoweil committed:
Revert: Unwanted code changes

shuoweil committed:
Revert "Revert: Unwanted code changes"

This reverts commit db5d8ea04ee3e8a6382ac546764aff0f6880f66b.

shuoweil committed:
revert 1 files to match main branch

shuoweil committed:
Correctly display DataFrames with JSON columns in anywidget

shuoweil committed:
Correctly display DataFrames with JSON columns in anywidget

shuoweil committed:
add mis-deleted comment back

shuoweil committed:
revert unnecessary change

shuoweil committed:
move helper function to dtypes.py

shuoweil committed:
revert unnecessary testcase change

shuoweil committed:
Improve JSON type handling for to_gbq and to_pandas_batches

shuoweil committed:
Remove unnecessary comment

shuoweil committed:
Merge branch 'main' into shuowei-anywidget-col

shuoweil committed:
Merge branch 'main' into shuowei-anywidget-col

shuoweil committed:
Revert bigframes/dtypes.py and mypy.ini to main branch version

tswast reviewed (approved):
Thanks!
